### PR TITLE
awx.awx.token: doc: show how token can be retrieved from facts

### PR DESCRIPTION
**Pull request by vrubiolo**
Signed-off-by: Vincent Rubiolo <vincent.libre@cryostase.eu>

##### SUMMARY
The documentation is mentioning the fact that the token is put into the fact system + refer to the examples but they do not show that. This PR corrects this.

##### ISSUE TYPE

 - Docs Pull Request

##### COMPONENT NAME

None (docs)

##### AWX VERSION

```
19.4.0
```

### CODE REVIEW

**Comment by vrubiolo** [comment]
This is my first PR, please let me know if I need to do something else. I just modified the docs inline, I have not tried to link them afterwards (but found no info about that either).

Edit: also wondering whether https://github.com/ansible/awx/blob/devel/awx_collection/plugins/modules/token.py#L77 should be changed too

File: awx_collection/plugins/modules/token.py
Comment (AlanCoding) [inline awx_collection/plugins/modules/token.py:90]: So this is the real functional change. Does it not work with just `token`? Is this just to kind of clarify where it comes from?

Comment (john-westcott-iv) [reply awx_collection/plugins/modules/token.py:90]: It looks like this functionality did actually change in commit 4add72b9d27116ddd0b5f478634a956e588ba8aa:
```
-        'tower_token': last_response,
+        'token': last_response,
```
However, we shouldn't need to reference it with ansible_facts. We should just be able to use `controller_token.token`.

Comment (john-westcott-iv) [reply awx_collection/plugins/modules/token.py:90]: Also, you shouldn't have to reference the `.token`, the controller_api.py module already compensates for that. See https://github.com/ansible/awx/blob/devel/awx_collection/plugins/module_utils/controller_api.py#L118

**Review by john-westcott-iv — CHANGES_REQUESTED** [review]
Just change `token` to `controller_token` and everything should work. See comments.

File: awx_collection/plugins/modules/token.py
Comment (vrubiolo) [reply awx_collection/plugins/modules/token.py:90]: Thanks to both of you for your comments, sorry this fell through the notifications cracks during the holiday.
I will refresh my PR according to what you said and resubmit.

Comment (vrubiolo) [reply awx_collection/plugins/modules/token.py:90]: Sorry for the lack of movement here, just to let you know that this is still on my todo list.

**Comment by tanujdevops** [comment]
> Just change `token` to `controller_token` and everything should work. See comments.

Hello @john-westcott-iv 
I'm Tanuj, I'm an undergraduate student, pursuing my Software Engineering Degree.
I'm just getting started with Open Source and I'd really appreciate if you can guide me a little so that I could help resolve this issue and submit a PR, and hence, make my first Open Source contribution!
Thank You!

File: awx_collection/plugins/modules/token.py
Comment (fosterseth) [reply awx_collection/plugins/modules/token.py:90]: @vrubiolo we are circling back to this, we are still interested in getting this doc change. Could you get a chance to make the change as described above? Once that in, we'll merge it

thanks

**Comment by fosterseth** [comment]
@tanujdevops thank you for your interest in contributing. We've marked some issues as "good first issue", check out this list https://github.com/ansible/awx/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22

Get a development environment set up https://github.com/ansible/awx/blob/devel/CONTRIBUTING.md
and dig in!

**Comment by TheRealHaoLiu** [comment]
@vrubiolo thank you for your contribution, AWX team is looking to pull this PR into the project. @fosterseth will be creating a new PR base on your commit and get it through our approval and test process.

Your contribution and co-authorship will be preserved.

btw have you heard about our community chatroom on Matrix? https://matrix.to/#/#awx:ansible.com come join us and a ton of other people that are passionate about AWX!

**Comment by john-westcott-iv** [comment]
Closing in favor of https://github.com/ansible/awx/pull/11550

**Pull request closed without merging**
